app/controllers/concerns/shopify_app/ensure_installed.rb

```diff
@@ -58,5 +63,14 @@ def shop_login
 
       url.to_s
     end
+
+    def validate_non_embedded_session
```

Comment:
Do you think calling this `validate_session_before_embedding` would make it easier to understand this method?